app/registries/foreman/plugin.rb

```diff
@@ -91,6 +103,7 @@ def unregister(plugin_id)
       # It doesn't unload installed plugins
       def clear
         @registered_plugins = {}
+        @registries = {}
```

Comment:
It doesn't undefine the registry methods. That means the plugin class will still respond to them, but return nil.

Reply:
solved by calling the initialize from clear, but now the method is redefined every time `clear` is called, but arguably it's fine, because that should never happen in Production environment as we currently don't have a mechanism to reload the plugin definition.